pkg/apis/core/validation/shoot.go

```diff
@@ -635,6 +637,27 @@ func validateNetworking(networking core.Networking, fldPath *field.Path) field.E
 	return allErrs
 }
 
+// ValidateWatchCacheSizes validates the given WatchCacheSizes fields.
+func ValidateWatchCacheSizes(sizes *core.WatchCacheSizes, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if sizes != nil {
+		if defaultSize := sizes.Default; defaultSize != nil && *defaultSize < 0 {
```

Comment:
You could use `apivalidation.ValidateNonnegativeField`.

Reply:
Done